src/models/msg.js

```diff
@@ -6,6 +6,15 @@ var id = 0;
 
 class Msg {
 	constructor(attr) {
+		// Some properties need to be copied in the Msg object instead of referenced
+		if (attr) {
+			["from", "invited", "target"].forEach((prop) => {
+				if (attr[prop]) {
+					this[prop] = _.cloneDeep(attr[prop]);
+				}
```

Comment:
As with previous PR, store only nick and mode because we don't want to copy all the other unnecessary stuff (away message etc)

Reply:
Fair enough. I figured it would be better for future sake to store the whole object, but I guess we don't have to at the moment.

Reply:
Done.